core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -411,10 +411,12 @@ class TeamPolicyInternal<Kokkos::Cuda, Properties...>
     KOKKOS_ASSERT(block_size > 0);
 
     // Currently we require Power-of-2 team size for reductions.
-    int p2 = 1;
-    while (p2 <= block_size) p2 *= 2;
-    p2 /= 2;
-    return p2 / impl_vector_length();
+    int pow2 = 0;
+    int block_size_pow2 = block_size;
+    while (block_size_pow2 >>= 1 > 0 ) pow2++;
+    block_size_pow2 = 1 << pow2;
+
+    return block_size_pow2 / impl_vector_length();
```

Comment:
Does this make any difference? AFAICT (https://godbolt.org/z/x6v1MEWb6) they show the same behavior.

Reply:
Change is for consistency following the added code block to compute pow2 in the ParallelReduce constructor. In case of using Kokkos::Auto in the ExecutionPolicy for that parallel_reduce, cuda_get_opt_block_size returns a non-pow2 team size. Setting the team size to the next smallest pow2, line 2007, fixes the reported issue.